llama-index-integrations/vector_stores/llama-index-vector-store-firestore/llama_index/vector_store/firestore/base.py

```diff
@@ -0,0 +1,218 @@
+        return "<"
+    if operator == FilterOperator.LTE:
+        return "<="
+    if operator == FilterOperator.IN:
+        return "in"
+    if operator == FilterOperator.NIN:
+        return "not-in"
+    if operator == FilterOperator.CONTAINS:
+        return "array-contains"
+
+    raise ValueError(f"Operator {operator} not supported in Firestore.")
+
+
+def _to_firestore_filter(
+    standard_filters: MetadataFilters,
+) -> Union[FieldFilter, Or, And, List[FieldFilter]]:
+    """Convert from standard dataclass to filter dict."""
+    firestore_filters = []
+
+    # Convert standard filters to Firestore filters
+    # Firestore filters are FieldFilter or And/Or filters
+    for f in standard_filters.filters:
+        if isinstance(f, MetadataFilter):
+            firestore_filters.append(
+                FieldFilter(f.key, _to_firestore_operator(f.operator), f.value)
+            )
+        elif isinstance(f, ExactMatchFilter):
+            firestore_filters.append(FieldFilter(f.key, "==", f.value))
+
+    if len(firestore_filters) == 1:
+        return firestore_filters[0]
+    if standard_filters.condition == FilterCondition.AND:
+        return And(filters=firestore_filters)
+    if standard_filters.condition == FilterCondition.OR:
+        return Or(filters=firestore_filters)
+
+    return firestore_filters
+
+
+class FirestoreVectorStore(BasePydanticVectorStore):
+    """Firestore Vector Store."""
+
+    stores_text: bool = True
+    flat_metadata: bool = True
+
+    collection_name: str
+    batch_size: Optional[int] = DEFAULT_BATCH_SIZE
+    embedding_key: Optional[str] = "embedding"
+    text_key: Optional[str] = "text"
+    metadata_key: Optional[str] = "metadata"
+    distance_strategy: Optional[DistanceMeasure] = DistanceMeasure.COSINE
+
+    _client: Client
+
+    def __init__(
+        self,
+        client: Optional[Client] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize params."""
+        super().__init__(**kwargs)
+        object.__setattr__(self, "_client", client or Client())
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "FirestoreVectorStore"
+
+    @property
+    def client(self) -> Any:
+        return self._client
+
+    def add(
+        self,
+        nodes: List[BaseNode],
+    ) -> List[str]:
+        """Add nodes to vector store."""
+        ids = []
+        entries = []
+        for node in nodes:
+            node_id = node.node_id
+            metadata = node_to_metadata_dict(
+                node,
+                remove_text=not self.stores_text,
+                flat_metadata=self.flat_metadata,
+            )
+            entry = {
+                self.embedding_key: node.get_embedding(),
+                self.metadata_key: metadata,
+            }
+            ids.append(node_id)
+            entries.append(entry)
+        self._upsert_batch(entries, ids)
+        return ids
+
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
```

Comment:
Does it complains if you remove the **delete_kwargs ?  (I assume yes)

Since its's unused

Reply:
Yes will complain about the signature not matching base method